src/studio/src/designer/frontend/ux-editor/components/rightMenu/PagesContainer.tsx

```diff
@@ -0,0 +1,288 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import { Button, Grid, IconButton, ListItemIcon, ListItemText, makeStyles, Menu, MenuItem, MenuProps, TextField, TextFieldProps, Typography, withStyles } from '@material-ui/core';
+import { getLanguageFromKey, getParsedLanguageFromKey } from 'app-shared/utils/language';
+import * as React from 'react';
+import { useSelector } from 'react-redux';
+import FormDesignerActionDispatchers from '../../actions/formDesignerActions/formDesignerActionDispatcher';
+import ConfirmModal from './ConfirmModal';
+
+export default function PagesContainer() {
+  const layoutOrder: string[] = useSelector((state: IAppState) => state.formDesigner.layout.layoutOrder);
+
+  return (
+    <Grid
+      container={true}
+    >
+      {layoutOrder.map((layout: string) => {
+        return (
+          <PageElement name={layout} key={layout}/>
+        );
+      })}
+    </Grid>
+  );
+}
+
+export interface IPageElementProps {
+  name: string;
+}
+
+const useStyles = makeStyles({
+  ellipsisButton: {
+    marginLeft: '1.2rem',
+    visibility: 'hidden',
+  },
+  mainButton: {
+    width: '100%',
+    fontSize: '1.4rem',
+    fontWeight: 400,
+    textTransform: 'unset',
+    '&:hover #ellipsis-button': {
+      visibility: 'visible',
+    },
+    '&:focus #ellipsis-button': {
+      visibility: 'visible',
+    },
+  },
+});
+
+export function PageElement({
```

Comment:
Consider splitting this file into multiple files, since there are quite a few components here

Reply:
Refactored.